server/src/controllers/christmas-tree.es6

```diff
@@ -153,36 +152,59 @@ christmasTree.create = (req, res) => {
     .then(permit => {
       const tcsAppID = vcapConstants.payGovAppId;
       const xmlData = paygov.getXmlForToken(req.body.forestAbbr, req.body.orgStructureCode, permit);
-      postPayGov(xmlData).then(xmlResponse => {
-        xml2jsParse(xmlResponse, function(err, result) {
-          if (!err) {
-            try {
-              const startOnlineCollectionResponse =
-                result['S:Envelope']['S:Body'][0]['ns2:startOnlineCollectionResponse'][0][
-                  'startOnlineCollectionResponse'
-                ][0];
-              const token = startOnlineCollectionResponse.token[0];
-              permit
-                .update({
-                  paygovToken: token
-                })
-                .then(savedPermit => {
-                  return res.status(200).json({
-                    token: token,
-                    permitId: savedPermit.permitId,
-                    payGovUrl: vcapConstants.payGovClientUrl,
-                    tcsAppID: tcsAppID
+      postPayGov(xmlData)
```

Comment:
this is a long string of promises - ok to break down a bit?

Reply:
Yes, moved to other functions!